lib/vector-vrl/tests/src/docs.rs

```diff
@@ -111,86 +111,84 @@ fn examples_to_tests(
     Box::new(examples.into_iter().flat_map(move |(k, v)| {
         v.examples
             .into_iter()
-            .map(|example| Test::from_cue_example(category, k.clone(), example))
+            .map(|example| test_from_cue_example(category, k.clone(), example))
             .collect::<Vec<_>>()
     }))
 }
 
-impl Test {
-    fn from_cue_example(category: &'static str, name: String, example: Example) -> Self {
-        use ::value::Value;
-
-        let Example {
-            title,
-            input,
-            mut source,
-            returns,
-            output,
-            raises,
-            skip_test,
-        } = example;
-
-        let mut skip = skip_test.unwrap_or_else(|| SKIP_FUNCTION_EXAMPLES.contains(&name.as_str()));
-
-        let object = match input {
-            Some(event) => {
-                serde_json::from_value::<Value>(serde_json::Value::Object(event.log)).unwrap()
-            }
-            None => Value::Object(BTreeMap::default()),
-        };
-
-        if returns.is_some() && output.is_some() {
-            panic!(
-                "example must either specify return or output, not both: {}/{}",
-                category, &name
-            );
-        }
+fn test_from_cue_example(category: &'static str, name: String, example: Example) -> Test {
```

Comment:
I'm curious why this method for creating a `Test` was moved out of `impl Test`.

Reply:
The `Test` struct is in a different crate (VRL) and this code is in Vector. Cue is Vector specific code.